ui/app/AppLayouts/Chat/ChatColumn/ChatComponents/SignTransactionModal.qml

```diff
@@ -68,7 +74,14 @@ ModalPopup {
         TransactionFormGroup {
             id: group1
             //% "Send"
-            headerText: qsTrId("command-button-send")
+            headerText: {
+                if(trxData.startsWith("0x095ea7b3")){
+                    const approveData = JSON.parse(walletModel.decodeTokenApproval(selectedRecipient.address, trxData))
+                    if(approveData.symbol)
+                        return qsTr("Authorize %1 %2").arg(approveData.amount).arg(approveData.symbol)    
+                }
+                return qsTrId("command-button-send");
+            }
```

Comment:
Where is the encoded data coming from? IOW, why is the gas not already estimated at the origin?

Reply:
The encoded data comes from any transaction created in the browser. Having a gas estimation will depend on whether the dapp developer did it or not. so.. :thinking:  we probably should check if the payload contains a gasLimit before calling estimateGas ourselves